docs/templates/eia860_child.rst.jinja

```diff
@@ -0,0 +1,74 @@
+{% extends "data_source_parent.rst.jinja" %}
+{% block background %}
+The Form EIA-860 collects utility, owner, plant, and generator-level data from existing
+and planned entities with one or more megawatt of capacity. The form also contains
+information regarding environmental control equipment and construction cost data from
+2013-2018.
+
+    * :download:`EIA-860 Instructions (PDF, to 2013-10-31) <{{ source.name }}/{{ source.name }}_instructions_2013-10-31.pdf>`
+    * :download:`EIA-860 Instructions (PDF, to 2017-05-31) <{{ source.name }}/{{ source.name }}_instructions_2017-05-31.pdf>`
+    * :download:`EIA-860 Instructions (PDF, to 2020-03-31) <{{ source.name }}/{{ source.name }}_instructions_2020-03-31.pdf>`
+    * :download:`EIA-860 Instructions (PDF, to 2023-05-31) <{{ source.name }}/{{ source.name }}_instructions_2020-03-31.pdf>`
```

Comment:
Given that these filenames & links can't be generated programmatically based on the source, I don't know that it makes sense to enumerate the particular filenames here and substitute in the source name. It doesn't really automate anything or avoid duplication and is a little obscure.

Is there an easy way to iterate over all the files (or all the PDFs, or all the filenames matching some pattern) in a directory, and use those paths / filenames to construct this list given the name of the data source? that way we could add additional blank forms / instructions by simply adding a new PDF to the docs dir.

Reply:
(and similarly in the other templates doing the same thing)

Reply:
Changed this to iterate through the directory for each source in `docs/data_sources` to get all of the downloadable PDFs. Then then the list of PDFs is dynamically generated in the RST.